test/unit/model/qti/datatype/LanguageValidateLanguageCodesTest.php

```diff
@@ -1,24 +1,101 @@
 <?php
 
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2023 (original work) Open Assessment Technologies SA.
+ */
+
 namespace oat\taoQtiItem\test\unit\model\qti\datatype;
 
-use oat\generis\test\TestCase;
+use PHPUnit\Framework\TestCase;
 use oat\taoQtiItem\model\qti\datatype\Language;
 
 class LanguageValidateLanguageCodesTest extends TestCase
 {
     private const LANGUAGE_CODES_JSON_PATH = __DIR__ . '/../../../../resources/pisa-qa-translation-locales.json';
 
     /**
-     * @dataProvider provideLanguageCodes
+     * @dataProvider validPisaLanguageCodesDataProvider
+     * @dataProvider validTwoLetterCodes
+     * @dataProvider invalidLanguageCodesDataProvider
      */
-    public function testValidate(string $languageCode): void
+    public function testValidate(bool $expected, string $languageCode): void
+    {
+        $this->assertEquals($expected, Language::validate($languageCode));
+    }
+
+    public function validPisaLanguageCodesDataProvider(): array
+    {
+        $languageData = json_decode(
+            file_get_contents(realpath(self::LANGUAGE_CODES_JSON_PATH)),
+            true
+        );
+
+        return array_map(
+            function(array $localeData) {
+                return [true, $localeData['code']];
+            },
+            $languageData
+        );
```

Comment:
We can use arrow function here 🙂 
```suggestion
        return array_map(
            static fn (array $localeData): array => [true, $localeData['code']],
            $languageData
        );
```